### PR TITLE
chore: fix aria issues in settings modal; fix and normalize focus ring usage

### PR DESCRIPTION
**Pull request by gribnoysup**
This fixes a few issues with the aria labels / ids and focusability of the tab content in the settings modal following the [example](https://www.w3.org/WAI/ARIA/apg/example-index/tabs/tabs-manual.html) and documentation I linked to in https://github.com/mongodb-js/compass/pull/3222#discussion_r912736402

I am deliberately not adding a proper keyboard support as this is more involved and for now, while we have only one tab there, can be skipped, but should be addressed as soon as we start adding more tabs there (I opened [a ticket](https://jira.mongodb.org/browse/COMPASS-6097) for that).

I also added a proper focus ring to the elements there and, while I was at it, normalized how we use it as there were a lot of different variations, some of which were just not working correctly

### CODE REVIEW

File: packages/compass-components/src/hooks/use-focus-ring.ts [hunk 1]
```diff
@@ -16,26 +15,112 @@ export const focusRingStyles = css({
...
+
+  const hoverClass = useMemo(() => {
+    return hover && focusRingHover;
+  }, [hover]);
 
   return {
-    ...focusProps,
     className: cx(
-      focusRingStyles,
-      focusState === FocusState.FocusVisible && focusRingVisibleStyles
```
Comment (gribnoysup) [inline packages/compass-components/src/hooks/use-focus-ring.ts:38]: Looking at the cases we have there is not really a point anymore to do this with js, can be just a `focus-visible` style instead

File: packages/compass-databases-navigation/src/databases-navigation-tree.tsx [hunk 2]
```diff
@@ -1,9 +1,13 @@
 /* eslint-disable react/prop-types */
 import React, { useCallback, useMemo, memo, useRef } from 'react';
-import { VisuallyHidden } from '@react-aria/visually-hidden';
```
Comment (gribnoysup) [inline packages/compass-databases-navigation/src/databases-navigation-tree.tsx:3]: Drive-by: we moved some other react-aria imports to components already

File: packages/compass-indexes/src/components/indexes-table/badge-with-icon-link.tsx [hunk 3]
```diff
@@ -23,9 +22,8 @@ const linkStyles = css(
       // LG uses backgroundImage instead of textDecoration
       backgroundImage: 'none !important',
     },
-    '&:focus': focusRingVisibleStyles,
```
Comment (gribnoysup) [inline packages/compass-indexes/src/components/indexes-table/badge-with-icon-link.tsx:26]: This should've been focus-visible

File: packages/compass-query-bar/src/components/option-editor.tsx [hunk 4]
```diff
@@ -16,34 +15,12 @@ import type { Ace } from 'ace-builds';
...
-      content: '""',
-      pointerEvents: 'none',
-      top: -1,
-      right: -1,
-      bottom: -1,
-      left: -1,
-      borderRadius: spacing[1],
-      transition: 'box-shadow .16s ease-in',
-      boxShadow: '0 0 0 0 transparent',
-    },
```
Comment (gribnoysup) [inline packages/compass-query-bar/src/components/option-editor.tsx:34]: This was overriding focus ring styles applied above

File: packages/compass-settings/src/components/modal.tsx [hunk 5]
```diff
@@ -95,7 +101,8 @@ export const SettingsModal: React.FunctionComponent<SettingsModalProps> = ({
           className={settingsStyles}
           data-testid="settings-modal-content"
           role="tabpanel"
-          id={`tabpanel-${selectedSetting}`}
+          tabIndex={0}
```
Comment (gribnoysup) [inline packages/compass-settings/src/components/modal.tsx:104]: It is suggested that tabpanels where first element is not focusable (like a link) are focusable, [see example](https://www.w3.org/WAI/ARIA/apg/example-index/tabs/tabs-manual.html)

File: packages/compass-settings/src/components/sidebar.tsx [hunk 6]
```diff
@@ -45,9 +50,9 @@ const SettingsSideNav: React.FunctionComponent<SidebarProps> = ({
           key={item}
           type="button"
           role="tab"
-          aria-controls={`${item} Tab`}
+          aria-controls={`${item} Section`}
```
Comment (gribnoysup) [inline packages/compass-settings/src/components/sidebar.tsx:53]: aria-controls should point to the panel, not to the button that shows the panel

File: packages/compass-sidebar/src/components/non-genuine-marker.tsx [hunk 7]
```diff
@@ -38,10 +37,9 @@ export default function NonGenuineMarker({
   return (
     <div className={nonGenuineMarkerContainer}>
       <button
-        {...focusRingProps}
```
Comment (gribnoysup) [inline packages/compass-sidebar/src/components/non-genuine-marker.tsx:41]: This would apply a class name that would be just overriden by the class name below

File: packages/compass-query-bar/src/components/query-option.tsx [hunk 8]
```diff
@@ -27,14 +26,11 @@ const queryOptionLabelStyles = css({
   marginRight: spacing[2],
 });
 
-const documentEditorOptionStyles = css(
-  {
-    minWidth: spacing[7],
-    display: 'flex',
-    flexGrow: 1,
-  },
-  focusRingStyles
```
Comment (gribnoysup) [inline packages/compass-query-bar/src/components/query-option.tsx:36]: This was just not doing anything

File: packages/compass-settings/src/components/modal.tsx [hunk 9]
```diff
@@ -82,7 +86,9 @@ export const SettingsModal: React.FunctionComponent<SettingsModalProps> = ({
       setOpen={closeModal}
       data-testid="settings-modal"
     >
-      <ModalTitle data-testid="settings-modal-title">Settings</ModalTitle>
+      <ModalTitle id="settings-tablist" data-testid="settings-modal-title">
```
Comment (gribnoysup) [inline packages/compass-settings/src/components/modal.tsx:89]: This id was used as aria-labeledby but wasn't provided

File: packages/compass-sidebar/src/components/non-genuine-marker.tsx (hunk 7 above)
Comment (lerouxb) [reply packages/compass-sidebar/src/components/non-genuine-marker.tsx:41]: are the props and behaviour of useFocusRing, etc not relevant here?

Comment (lerouxb) [reply packages/compass-sidebar/src/components/non-genuine-marker.tsx:41]: Oh I see you changed useFocusRing as well anyway. Nevermind, then.

Comment (gribnoysup) [reply packages/compass-sidebar/src/components/non-genuine-marker.tsx:41]: Not anymore, it just returns a `className` now, [see here](https://github.com/mongodb-js/compass/pull/3424/files#diff-57f6739adcc3ac73c61e6882a02f5198a3bd269d5f5fe6ffb7f487cfa42f1bc3R79).

And for some additional context the reason for still having a hook is that in a lot of places the way just the css part was used was incorrect leading to a non-working focus ring, so when some special behavior is needed from the focus ring, the hook does the css merging in one place so that we can control better how it happens to avoid errors

Comment (gribnoysup) [reply packages/compass-sidebar/src/components/non-genuine-marker.tsx:41]: Posted this before I saw your last comment, sorry 🙈

**Review by lerouxb — APPROVED** [review]
